compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -177,10 +177,9 @@ class TypeApplications(val self: Type) extends AnyVal {
       case self: TypeRef =>
         val tsym = self.symbol
         if (tsym.isClass) tsym.typeParams
-        else if (!tsym.exists || !isTrivial(self.prefix, tsym)) self.info.typeParams
         else tsym.infoOrCompleter match {
-          case info: LazyType => info.completerTypeParams(tsym)
-          case info => info.typeParams
+          case info: LazyType if tsym.exists && isTrivial(self.prefix, tsym) => info.completerTypeParams(tsym)
+          case _ => self.info.typeParams
```

Comment:
Why change this line from `case info => info.typeParams`? If the info is already computed I would just reuse it

Reply:
To clarify (for @allanrenucci and future readers): if `tsym.exists && isTrivial(self.prefix, tsym)`, `self.info.typeParams` and `info.typeParams` should coincide and `self.info.typeParams` is more clearly right, so I tried to simplify things.

Reply:
Actually I tried: `info.typeParams` and `self.info.typeParams` are not equivalent, the newly added test was failing. I think this is because in the former, `info` may refer to the completer

Reply:
They’re only equivalent under those conditions, not in this pattern match branch.
I know that my commit didn’t change behavior in any scenario I tested with.
To play with this, you can go to the parent commit and replace info.typeParams with self.info.typeParams.